### PR TITLE
Dashboard architecture overview documentation

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
This PR adds overview documentation for the dashboard architecture.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

**Comment by petersutter** [comment]
please highlight the components with a code block, like `backed`, `frontend`, `kube-apiserver` etc

File: docs/architecture/README.md [hunk 1]
```diff
@@ -2,16 +2,31 @@
...
 
 <img src="../images/architecture-2.png">
 
 ## Backend
+The backend is currently a monolithic Node.js application, but it performs several tasks that are actually independent.
+
+* Static web server for the frontend single page application
+* Forward real time events of the apiserver to the frontend
+* Provide a HTTP Api 
+* Bootstrapping shoot and seed clusters to support web terminals
```
Comment (petersutter) [inline docs/architecture/README.md:20]: link to https://github.com/gardener/dashboard/blob/master/docs/concepts/webterminals.md#bootstrapping documentation

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**